runtime(vim): Update ftplugin, fix option variable 'keywordprg' matching

dkearns opened the pull request:
- Match &option, and &[lg]:option variables.
- Match commands after :bar.
- Fix matching of pre and post context text.
- Style - use '..' for string concatenation.

Fixes issue #17567.

Konfekt commented on runtime/ftplugin/vim.vim:
```diff
@@ -56,41 +56,42 @@ if !exists("*" .. expand("<SID>") .. "Help")
-      return 'v'.matchstr(post, ':\w\+')
+      return 'v' .. matchstr(post, ':\w\+')
+    elseif pre =~# '&\%([lg]:\)\='
+      return "'" .. topic "'"
```
How come double dots dispensable on the right-hand side here?

dkearns replied:
They're not, it's a typo.  Thanks, for noticing.

The second string is being parsed as a tail comment and `:help 'option` generally works just as well as `:help 'option'` so I hadn't noticed.  The highlighting also doesn't match that as a comment.

I'll fix both issues, thanks.

dkearns committed:
runtime(vim): Update ftplugin, fix option variable 'keywordprg' matching

- Match &option, and &[lg]:option variables.
- Match Ex commands after :bar.
- Fix matching of pre and post context text.
- Style - use '..' for string concatenation.

Fixes issue #17567.
￼
Signed-off-by: Doug Kearns <dougkearns@gmail.com>

zeertzjq commented on runtime/ftplugin/vim.vim:
```diff
@@ -1,8 +1,9 @@
 " Vim filetype plugin
 " Language:          Vim
 " Maintainer:        Doug Kearns <dougkearns@gmail.com>
+" Last Change:       2025 Aug 07
```
There is already a "Last Change" line below. One of them should be removed.
